actors/builtin/miner/miner_test.go

```diff
@@ -390,6 +390,112 @@ func TestProvingPeriodCron(t *testing.T) {
 	})
 }
 
+func TestExtendSectorExpiration(t *testing.T) {
+	owner := tutil.NewIDAddr(t, 100)
+	worker := tutil.NewIDAddr(t, 101)
+	workerKey := tutil.NewBLSAddr(t, 0)
+	actor := newHarness(t, tutil.NewIDAddr(t, 1000), owner, worker, workerKey)
+	periodOffset := abi.ChainEpoch(100)
+	builder := mock.NewBuilder(context.Background(), actor.receiver).
+		WithActorType(owner, builtin.AccountActorCodeID).
+		WithActorType(worker, builtin.AccountActorCodeID).
+		WithHasher(fixedHasher(uint64(periodOffset))).
+		WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("rejects negative extension", func(t *testing.T) {
+		rt, _, _, sector := setupExtension(t, builder, actor, periodOffset)
+
+		// attempt to shorten epoch
+		newExpiration := sector.Info.Expiration - abi.ChainEpoch(miner.WPoStProvingPeriod)
+		params := &miner.ExtendSectorExpirationParams{
+			SectorNumber:  sector.Info.SectorNumber,
+			NewExpiration: newExpiration,
+		}
+
+		func() {
+			defer func() {
+				r := recover()
+				require.NotNil(t, r)
+				assert.Contains(t, fmt.Sprintf("%s",r), "cannot reduce sector expiration")
+			}()
+
+			actor.extendSector(rt, sector, 0, params)
+		}()
+	})
+
+	t.Run("rejects extention to invalid epoch", func(t *testing.T) {
+		rt, _, _, sector := setupExtension(t, builder, actor, periodOffset)
+
+		// attempt to extend to an epoch that is not a multiple of the proving period + the commit epoch
+		extension := 42*miner.WPoStProvingPeriod + 1
+		newExpiration := sector.Info.Expiration - abi.ChainEpoch(extension)
+		params := &miner.ExtendSectorExpirationParams{
+			SectorNumber:  sector.Info.SectorNumber,
+			NewExpiration: newExpiration,
+		}
+
+		func() {
+			defer func() {
+				r := recover()
+				require.NotNil(t, r)
+				assert.Contains(t, fmt.Sprintf("%s",r), "invalid expiration")
+			}()
+
+			actor.extendSector(rt, sector, uint64(extension), params)
+		}()
```

Comment:
Use `rt.ExpectAbort`.

As my comment in #435 , combine this with other test cases to reduce the set-up.